### PR TITLE
INTYGFV-15374: Added statement for Show personal data in patient view

### PR DESCRIPTION
**Pull request by UlrikaSopra**

### CODE REVIEW

**Review by SamuelTilly — CHANGES_REQUESTED** [review]
Very nice, this could use some tests to make sure we display information correctly.

**Comment by UlrikaSopra** [comment]
> Very nice, this could use some tests to make sure we display information correctly.

working on it :)

File: apps/rehabstod/src/pages/Patient/components/PatientHeader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      slut: '2023-05-01',
+      patient: {
+        ...patient.sjukfallList[0].intyg[0].patient,
+        alder: 30,
+        kon: 'Kvinna',
+        namn: 'Olle',
+        id: '123',
+      },
+    }
+  })
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/PatientHeader.test.tsx:36]: We can make this a bit simpler, but i noticed an issue with how our faker package handles mergers.

```tsx
function renderPatientHeader(showPersonalInformation: boolean) {
  const patient = fakePatient({
    sjukfallList: [
      {
        intyg: [
          {
            start: '2023-04-25',
            slut: '2023-05-01',
            patient: {
              alder: 30,
              kon: 'Kvinna',
              namn: 'Olle',
              id: '123',
            },
          },
        ],
      },
    ],
  })
  store.dispatch(updateShowPersonalInformation(showPersonalInformation))
  renderWithRouter(<PatientHeader patient={patient} />)
}
```

File: apps/rehabstod/src/utils/fake/fakePatient.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { GenerateMockOptions } from '@anatine/zod-mock'
+import { patientSchema, patientSjukfallSchema } from '../../schemas/patientSchema'
+
+const fakePatientSjukfall = fakerFromSchema(patientSjukfallSchema)
+
+export const fakePatient = fakerFromSchema(patientSchema, {
+  sjukfallList: Array.from({ length: faker.datatype.number({ min: 1, max: 5 }) }, fakePatientSjukfall),
+  srsError: faker.datatype.boolean(),
+  kompletteringInfoError: faker.datatype.boolean(),
+} as Omit<GenerateMockOptions, 'sjfMetaData'>)
```
Comment (SamuelTilly) [inline apps/rehabstod/src/utils/fake/fakePatient.ts:11]: We can use `fakerFromSchemaFactory` to wrap our faker in another layer, this avoids the typescript issue.

```typescript
export const fakePatient = fakerFromSchemaFactory(patientSchema, {
  sjukfallList: Array.from({ length: faker.datatype.number({ min: 1, max: 5 }) }, fakePatientSjukfall),
  srsError: faker.datatype.boolean(),
  kompletteringInfoError: faker.datatype.boolean(),
})
```

**Review by SamuelTilly — CHANGES_REQUESTED** [review]
We can make it a bit simpler to use faker functions. In addition to the things I list, we need to update the `deepmerge` function in `packages\fake\src\index.ts` so that it doesn't merge arrays.

```typescript
const customDeepmerge = deepmergeCustom<{
  DeepMergeArraysURI: DeepMergeLeafURI
}>({
  mergeArrays: false,
})

export function fakerFromSchema<T extends ZodTypeAny>(schema: T, options?: GenerateMockOptions) {
  return (data?: DeepPartial<z.infer<T>>) => customDeepmerge(generateMock(schema, { stringMap, faker, ...options }), data ?? {})
}
```

**Comment by UlrikaSopra** [comment]
> We can make it a bit simpler to use faker functions. In addition to the things I list, we need to update the `deepmerge` function in `packages\fake\src\index.ts` so that it doesn't merge arrays.
> 
> ```ts
> const customDeepmerge = deepmergeCustom<{
>   DeepMergeArraysURI: DeepMergeLeafURI
> }>({
>   mergeArrays: false,
> })
> 
> export function fakerFromSchema<T extends ZodTypeAny>(schema: T, options?: GenerateMockOptions) {
>   return (data?: DeepPartial<z.infer<T>>) => customDeepmerge(generateMock(schema, { stringMap, faker, ...options }), data ?? {})
> }
> ```

**Review by SamuelTilly — APPROVED** [review]

**Pull request merged**